src/Button/styles.ts

```diff
@@ -170,96 +172,140 @@ export const getVariantStyles = (variant: VariantType = 'default', theme?: Theme
-  }
-  return {
-    paddingY: `${paddingY}px`,
-    paddingX: `${paddingX}px`,
-    fontSize,
-    '[data-component=ButtonCounter]': {
-      fontSize,
-    },
-  }
-}
-
 export const getBaseStyles = (theme?: Theme) => ({
   borderRadius: '2',
   border: '1px solid',
   borderColor: theme?.colors.btn.border,
   fontFamily: 'inherit',
-  fontWeight: 'bold',
-  lineHeight: TEXT_ROW_HEIGHT,
-  whiteSpace: 'nowrap',
-  verticalAlign: 'middle',
+  fontWeight: 'semibold',
+  fontSize: '1',
   cursor: 'pointer',
   appearance: 'none',
   userSelect: 'none',
   textDecoration: 'none',
   textAlign: 'center',
-  '&:disabled': {
-    cursor: 'default',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  height: '32px',
+  padding: '0 12px',
+  gap: '8px',
+  minWidth: 'max-content',
+  transition: '80ms cubic-bezier(0.65, 0, 0.35, 1)',
+  transitionProperty: 'color, fill, background-color, border-color',
+  '&[href]': {
+    display: 'inline-flex',
+    '&:hover': {
+      textDecoration: 'none',
+    },
+  },
+  '&:hover': {
+    transitionDuration: '80ms',
   },
-  '&:disabled svg': {
-    opacity: '0.6',
+  '&:active': {
+    transition: 'none',
+  },
+  '&:disabled': {
+    cursor: 'not-allowed',
+    boxShadow: 'none',
   },
   '@media (forced-colors: active)': {
     '&:focus': {
       // Support for Windows high contrast https://sarahmhigley.com/writing/whcm-quick-tips
       outline: 'solid 1px transparent',
     },
   },
+  '[data-component=ButtonCounter]': {
+    fontSize: '1',
+  },
+  '&[data-component=IconButton]': {
+    display: 'inline-grid',
+    padding: 'unset',
+    placeContent: 'center',
+    width: '32px',
+  },
+  '&[data-size="small"]': {
+    padding: '0 8px',
+    height: '28px',
+    gap: '4px',
+    fontSize: '0',
+
+    '[data-component="text"]': {
+      lineHeight: 'calc(20 / 12)',
+    },
+
+    '[data-component=ButtonCounter]': {
+      fontSize: '0',
+    },
+
+    '[data-component="buttonContent"] > :not(:last-child)': {
+      mr: '4px',
+    },
+
+    '&[data-component=IconButton]': {
+      width: '28px',
+      padding: 'unset',
+    },
+  },
+  '&[data-size="large"]': {
+    padding: '0 16px',
+    height: '40px',
+    gap: '8px',
+
+    '[data-component="buttonContent"] > :not(:last-child)': {
+      mr: '8px',
+    },
+
+    '&[data-component=IconButton]': {
+      width: '40px',
+      padding: 'unset',
+    },
+  },
 })
 
 export const getButtonStyles = (theme?: Theme) => {
   const styles = {
     ...getBaseStyles(theme),
-    display: 'grid',
-    gridTemplateAreas: '"leadingIcon text trailingIcon"',
-    '& > :not(:last-child)': {
-      mr: '2',
+    '&[data-component="block"]': {
+      width: '100%',
     },
-    '[data-component="leadingIcon"]': {
-      gridArea: 'leadingIcon',
+    '[data-component="leadingVisual"]': {
+      gridArea: 'leadingVisual',
     },
     '[data-component="text"]': {
       gridArea: 'text',
+      lineHeight: 'calc(20/14)',
```

Comment:
This is a much nicer way to read line heights! 😍

Reply:
You can thank Vini for that one ❤️